lib/jxl/enc_ac_strategy.cc

```diff
@@ -487,12 +518,15 @@ float EstimateEntropy(const AcStrategy& acs, size_t x, size_t y,
     // bias.
     entropy += config.zeros_mul * (CeilLog2Nonzero(nbits + 17) + nbits);
   }
-  float ret =
-      entropy +
-      masking *
-          ((config.info_loss_multiplier * GetLane(SumOfLanes(df, info_loss))) +
-           (config.info_loss_multiplier2 *
-            sqrt(num_blocks * GetLane(SumOfLanes(df, info_loss2)))));
+  const float kMixLoss = kMixLossTable[acs.RawStrategy()];
+  const float loss1 = GetLane(SumOfLanes(df, info_loss));
+  const float loss2 = sqrt(GetLane(SumOfLanes(df, info_loss2)) * (num_blocks * 64));
+  const float loss = kMixLoss * (config.info_loss_multiplier * loss1) +
+               (1.0 - kMixLoss) * (config.info_loss_multiplier2 * loss2);
+  const float kRegulateSurface = 11.5f;
+  float large_surface_error_mul =
+      (kRegulateSurface + sqrt(num_blocks)) * (1.0f / (kRegulateSurface + 1));
+  float ret = entropy + large_surface_error_mul * masking * loss;
   return ret;
```

Comment:
In the future, I would consider getting rid of this intermediary `ret` variable.

https://abseil.io/tips/161

Reply:
thank you -- fixing it here